server/public/model/utils.go

```diff
@@ -335,22 +335,24 @@ func (er *AppError) Wrap(err error) *AppError {
 	return er
 }
 
-// AppErrorFromJSON will decode the input and return an AppError
-func AppErrorFromJSON(data io.Reader) *AppError {
-	str := ""
-	bytes, rerr := io.ReadAll(data)
-	if rerr != nil {
-		str = rerr.Error()
-	} else {
-		str = string(bytes)
+// AppErrorFromJSON will try to decode the input into an AppError.
+func AppErrorFromJSON(r io.Reader) error {
+	data, err := io.ReadAll(r)
+	if err != nil {
+		return err
 	}
 
-	decoder := json.NewDecoder(strings.NewReader(str))
 	var er AppError
-	err := decoder.Decode(&er)
+	err = json.NewDecoder(bytes.NewReader(data)).Decode(&er)
 	if err != nil {
-		return NewAppError("AppErrorFromJSON", "model.utils.decode_json.app_error", nil, "body: "+str, http.StatusInternalServerError).Wrap(err)
+		// If the request exceeded FileSettings.MaxFileSize a plain error gets returned. Convert it into an AppError.
+		if string(data) == "http: request body too large\n" {
```

Comment:
Wondering if we could spot this case earlier by looking at the expected HTTP status code instead of calling `AppErrorFromJSON` and have to resort to string matching.

Reply:
While this idea is interesting, I don't think it works here. The server returns a fairly generic 400 status code. Given that the same status code is also returned for e.g. semantic errors in the request, I don't think it's an option here.

Reply:
Interesting. Wondering if we could at least match against [`MaxBytesError.Error()`](https://pkg.go.dev/net/http#MaxBytesError.Error) but maybe even that's tricky.

Reply:
The test actually matches against `MaxBytesError` (https://github.com/mattermost/mattermost/blob/559c09d184758577c2c15bde4f397d5ebb8dda79/server/public/model/utils_test.go#L187-L188). I'm 0/5 if the actual implementation should also do that.